lib/ex_doc/formatter/html.ex

```diff
@@ -132,12 +132,41 @@ defmodule ExDoc.Formatter.HTML do
   end
 
   defp generate_list(nodes, all, output, config, has_readme) do
-    Enum.each nodes, &generate_module_page(&1, all, output, config, has_readme)
+    if config.main do
+      # In this case the call must be sync because the `generate_main` function
+      # depends on this result. This situation only applies when the option
+      # `-m` or `--main` is given.
+      nodes
+      |> Enum.map(&generate_module_page_sync(&1, all, output, config, has_readme))
+      |> Enum.map(fn(_) ->
+          receive do
+            {:ok, file} -> file
+          end
+        end)
+    else
+      nodes
+      |> Enum.map(&generate_module_page_async(&1, all, output, config, has_readme))
+    end
+  end
+
+  defp generate_module_page_async(node, modules, output, config, has_readme) do
+    spawn(fn ->
+      generate_module_page(node, modules, output, config, has_readme)
+    end)
+  end
```

Comment:
We need to wait for the result somewhere right? Otherwise we can start generating and then close the app without it finishing.